test/helpers/TemplateTests.h

```diff
@@ -0,0 +1,46 @@
+/**
+ * \file TemplateTests.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief     TODO
+ */
+
+
+#ifndef TEST_HELPERS_TEMPLATETESTS_H_
+#define TEST_HELPERS_TEMPLATETESTS_H_
+
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <utility>
+
+
+namespace Mercury {
+namespace test {
+
+
+/**
+ * Templated test to check that users of a base class can use it through a
+ * pointer of an interface it implements.
+ */
+template<class Base, class Interface, typename... Args>
+void testBaseIsAnInterface(Args&&... args) {
```

Comment:
Doesn't actually test if the base is an interface, just that the class can be instantiated as an interface pointer - function name could get a bit confusing.

Reply:
True, but then again you'd get a compile time error if you were to pass anything else in. It's true, I am technically testing two things with one here 🤔